frontend/src/views/dev_frame/load-mock-data.tsx

```diff
@@ -263,6 +275,126 @@ export function SeedDataMenu({
+                email: application.email,
+                phone: application.phone.toString(),
+                program: application.program,
+                department: application.department,
+                yip: application.yip,
+                gpa: application.gpa || 0,
+                previous_department_ta: application.previous_department_ta,
+                previous_university_ta: application.previous_university_ta,
+                program_start: application.program_start,
+                previous_other_university_ta:
+                    application.previous_other_university_ta,
+                position_preferences: application.position_preferences,
+            };
+
+            await apiPOST(
+                `/public/postings/${url_token}/submit`,
+                { answers: newApp },
+                true
+            );
+
+            count++;
+            setProgress(Math.round((count / total) * 100));
+        }
+
+        await dispatch(
+            debugOnlySetActiveUser({
+                utorid: initialUser.utorid,
+                roles: initialUser.roles,
+            })
+        );
+
+        setProgress(100);
+    }
+
+    async function seedInstructorPreferences(limit = 600) {
+        setStage("Instructor Preferences");
+        setProgress(0);
+
+        if (!targetSession) {
+            throw new Error("Need a valid session to continue");
+        }
+
+        let count = 0;
+        let total = seedData.applications.length;
+
+        for (const application of seedData.applications.slice(0, limit)) {
+            if (application.instructor_preferences) {
+                const targetApplication =
+                    applications.find(
+                        (currApplication) =>
+                            currApplication.applicant.utorid ===
+                            application.utorid
+                    ) || null;
+
+                if (!targetApplication) {
+                    throw new Error("No application found for " + application);
+                }
+
+                for (const position of application.instructor_preferences) {
+                    const targetPosition =
+                        positions.find(
+                            (currPosition) =>
+                                currPosition.position_code ===
+                                position.position_code
+                        ) || null;
+
+                    if (!targetPosition) {
+                        throw new Error(
+                            "No position found for " + position.position_code
+                        );
+                    }
+
+                    await apiPOST(`/instructor/instructor_preferences`, {
+                        preference_level: position.preference_level,
+                        comment: position.comment,
+                        application_id: targetApplication.id,
+                        position_id: targetPosition.id,
+                    });
```

Comment:
I feel weird about this loop. I think you should create an array of what needs to be upserted, and then, in a separate loop upsert the data via async calls.